CONTRIBUTING.md

```diff
@@ -1,3 +1,540 @@
+
+## Setting up the VirtualEnv
+
+Lets create a virtualenv and install the required packages so that we can start developing on Syft.
+
+### Pipenv
+
+Using pipenv you would do the following:
+
+```
+$ pipenv --python=3.9
+```
+
+We installed python 3.9 earlier so here we can just specify the version and we will get a virtualenv with that version. If you want to use a different version make sure to install it to your system with your system package manager or `pyenv` first.
+
+We have created the virtualenv but it is not active yet.
+If you type the following:
+
+```
+$ which python
+/usr/bin/python
+```
+
+You can see that we still have a python path that is in our system binary folder.
+
+Lets activate the virtualenv with:
+
+```
+$ pipenv shell
+```
+
+You should now see that the prompt has changed and if you run the following:
+
+```
+$ which python
+/Users/madhavajay/.local/share/virtualenvs/PySyft-lHlz_cKe/bin/python
+```
+
+Okay, any time we are inside the virtualenv every python and pip command we run will use this isolated version that we defined and will not affect the rest of the system or other projects.
+
+### Install Python Dependencies
+
+Once you are inside the virtualenv you can do this with pip or pipenv.
+
+#### Windows
+
+To install the dependencies properly on Windows, you must first install PyTorch because most Windows binary wheels are not available on PyPI.
+
+You can do this by telling `pip` to use the official PyTorch Wheel Repository instead of the default PyPI repository.
+
+```
+$ pip install torch torchvision -f https://download.pytorch.org/whl/torch_stable.html
+```
+
+_Note_ If you need a specific version you can supply it with the `==` syntax. Also be aware there are different versions depending on if you require CUDA for GPU usage or CPU only such as what we use in GitHub CI.
+
+```
+$ pip install torch==1.7.1 torchvision==0.8.2 -f https://download.pytorch.org/whl/torch_stable.html
+```
+
+Then continue below with requirements.dev.txt like normal.
+
+**NOTE** this is required for several `dev` packages like pytest-xdist etc.
+
+```
+$ pip install -r requirements.dev.txt
+```
+
+or
+
+```
+$ pipenv install --dev --skip-lock
+```
+
+Now you can verify we have installed a lot of stuff by running:
+
+```
+$ pip freeze
+```
+
+### Linking the SyMPC src
+
+Now we need to link the src directory of the SyMPC code base into our site-packages
+so that it acts like it’s installed but we can change any file we like and `import` again
+to see the changes.
+
+```
+$ pip install -e .
+```
+
+The best way to know everything is working is to run the tests.
+
+Run the quick tests with all your CPU cores by running:
+
+```
+$ pytest -n auto
+```
+
+If they pass then you know everything is set up correctly.
+
+
+### Linking the PySyft src
+
+SyMPC is a companion library for PySyft. For the moment they are highly coupled.
+One of the SyMPC goals is to convert it into a standalone library. Thus, PySyft
```

Comment:
Q: I think SyMPC depends on PySyft, right?

Reply:
I say it explicitly now. Is it better now?